ietf/meeting/views.py

```diff
@@ -760,31 +785,44 @@ def prepare_timeslots_for_display(timeslots, rooms):
                     timeslot=timeslot,
                 )
 
-            r = {'success': True}
             if tombstone_session:
                 prepare_sessions_for_display([tombstone_session])
-                r['tombstone'] = render_to_string("meeting/edit_meeting_schedule_session.html", {'session': tombstone_session})
-            return JsonResponse(r)
+                return _json_response(
+                    True,
+                    tombstone=render_to_string("meeting/edit_meeting_schedule_session.html",
+                                               {'session': tombstone_session})
+                )
+            else:
+                return _json_response(True)
 
         elif action == 'unassign' and request.POST.get('session', '').isdigit():
             session = get_object_or_404(sessions, pk=request.POST['session'])
-            SchedTimeSessAssignment.objects.filter(session=session, schedule=schedule).delete()
+            existing_assignments = SchedTimeSessAssignment.objects.filter(session=session, schedule=schedule)
+            assertion('len(existing_assignments) <= 1',
+                      note='Multiple assignments for {} in schedule {}'.format(session, schedule))
+            if not all(timeslot_locked(ea.timeslot) for ea in existing_assignments):
+                existing_assignments.delete()
```

Comment:
There should not actually be multiple assignments for a session within a single schedule, so something is already borked if this makes a difference. The choice here was to avoid getting stuck in an "impossible" situation, but probably better if they have to ask for admin help so I'll change the all to any.